scipy/stats/tests/test_qmc.py

```diff
@@ -317,6 +322,45 @@ def test_primes(self):
         out = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]
         assert_allclose(primes, out)
 
+    def test_lloyd(self):
+        def l1_norm(sample):
+            l1 = cdist(sample, sample, 'cityblock')
+            return np.min(l1[l1.nonzero()])
+
+        # mindist
+        def l2_norm(sample):
+            l2 = cdist(sample, sample)
+            return np.min(l2[l2.nonzero()])
+
+        # quite sensible seed as it can go up before going further down
+        rng = np.random.RandomState(1809831)
+        sample = rng.uniform(0, 1, size=(128, 2))
```

Comment:
One thing to think about is pathological input that violates *general position* (basically, invalidates most common Voronoi algorithms). Here is a classic example that will fail this test because of the mess it causes inside `Voronoi`:

```diff
diff --git a/scipy/stats/tests/test_qmc.py b/scipy/stats/tests/test_qmc.py
index a5346d885..de1e18a15 100644
--- a/scipy/stats/tests/test_qmc.py
+++ b/scipy/stats/tests/test_qmc.py
@@ -332,9 +332,13 @@ class TestUtils:
             l2 = cdist(sample, sample)
             return np.min(l2[l2.nonzero()])
 
-        # quite sensible seed as it can go up before going further down
-        rng = np.random.RandomState(1809831)
-        sample = rng.uniform(0, 1, size=(128, 2))
+        # test a common violation of general
+        # position for Voronoi generators
+        # by feeding in 4 co-circular generators
+        sample = np.array([[0.1, 0.1],
+                           [0.1, 0.2],
+                           [0.2, 0.1],
+                           [0.2, 0.2]])
         base_l1 = l1_norm(sample)
         base_l2 = l2_norm(sample)

```

```python
scipy/stats/tests/test_qmc.py:353: in test_lloyd
    assert base_l1 < curr_l1
E   assert 0.1 < 0.08999999998967997

```

I'm not exactly sure what we'd want to do here--maybe just giving the user the ability to feed in Qhull options as opposed to trying to detect co-circular points in n-dimensions. And/or a brief note in the documentation perhaps.

Reply:
Although this case is less obviously pathological:

```diff
+        sample = np.array([[0.9, 0.1],
+                           [0.1, 0.2],
+                           [0.29, 0.91],
+                           [0.7, 0.6],
+                           [0.3, 0.9]])
```

Maybe small numbers of points also present big challenges--the result is within precision to the same perhaps:

```
scipy/stats/tests/test_qmc.py:354: in test_lloyd
    assert base_l1 < curr_l1
E   assert 0.020000000121988992 < 0.01999999998787466
```

Reply:
`qhull_options` allows the user to pass in Qhull options, I think. Do you think this is now adequately addressed in the documentation, @tupui?